Flaky test failure: test_single_qubit_matrix_to_native_gates_cases

This random test failure was observed on travis ( https://travis-ci.com/quantumlib/Cirq/jobs/228700099 , but likely no longer appears due to re-running the check ):

```
_______ test_single_qubit_matrix_to_native_gates_cases[intended_effect9] _______
intended_effect = array([[-0.34715252+0.04111902j, -0.84537193+0.40390674j],
       [ 0.91630224+0.19540869j, -0.34716016+0.04105442j]])
    @pytest.mark.parametrize('intended_effect', [
        np.array([[0, 1j], [1, 0]]),
    ] + [
        cirq.testing.random_unitary(2) for _ in range(10)
    ])
    def test_single_qubit_matrix_to_native_gates_cases(intended_effect):
        gates = cirq.single_qubit_matrix_to_phased_x_z(
            intended_effect, atol=0.0001)
        assert len(gates) <= 2
>       assert_gates_implement_unitary(gates, intended_effect, atol=1e-8)
cirq/optimizers/decompositions_test.py:241: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
cirq/optimizers/decompositions_test.py:32: in assert_gates_implement_unitary
    atol=atol)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
actual = array([[-0.33096274+1.12558019e-01j, -0.74264474+5.71203247e-01j],
       [ 0.9369068 -2.77555756e-17j, -0.33096274+1.12558019e-01j]])
desired = array([[-0.33094179+0.1126196j , -0.74253844+0.57134143j],
       [ 0.9369068 +0.j        , -0.33096274+0.11255802j]])
    def assert_allclose_up_to_global_phase(
            actual: np.ndarray,
            desired: np.ndarray,
            *,  # Forces keyword args.
            rtol: float = 1e-7,
            atol: float,  # Require atol to be specified
            equal_nan: bool = True,
            err_msg: Optional[str] = '',
            verbose: bool = True) -> None:
        """Checks if a ~= b * exp(i t) for some t.
    
        Args:
            actual: A numpy array.
            desired: Another numpy array.
            rtol: Relative error tolerance.
            atol: Absolute error tolerance.
            equal_nan: Whether or not NaN entries should be considered equal to
                other NaN entries.
            err_msg: The error message to be printed in case of failure.
            verbose: If True, the conflicting values are appended to the error
                message.
    
        Raises:
            AssertionError: The matrices aren't nearly equal up to global phase.
        """
        actual, desired = linalg.match_global_phase(actual, desired)
        np.testing.assert_allclose(
            actual=actual,
            desired=desired,
            rtol=rtol,
            atol=atol,
            equal_nan=equal_nan,
            err_msg=err_msg,
>           verbose=verbose)
E       AssertionError: 
E       Not equal to tolerance rtol=1e-07, atol=1e-08
E       
E       Mismatch: 50%
E       Max absolute difference: 0.00017434
E       Max relative difference: 0.00018608
E        x: array([[-0.330963+1.125580e-01j, -0.742645+5.712032e-01j],
E              [ 0.936907-2.775558e-17j, -0.330963+1.125580e-01j]])
E        y: array([[-0.330942+0.11262j , -0.742538+0.571341j],
E              [ 0.936907+0.j      , -0.330963+0.112558j]])
cirq/testing/lin_alg_utils.py:139: AssertionError
```

--- a/cirq/optimizers/decompositions_test.py
+++ b/cirq/optimizers/decompositions_test.py
@@ -236,9 +236,9 @@ def test_single_qubit_matrix_to_native_gates_known():
 ])
 def test_single_qubit_matrix_to_native_gates_cases(intended_effect):
     gates = cirq.single_qubit_matrix_to_phased_x_z(
-        intended_effect, atol=0.0001)
+        intended_effect, atol=1e-6)
     assert len(gates) <= 2
-    assert_gates_implement_unitary(gates, intended_effect, atol=1e-8)
+    assert_gates_implement_unitary(gates, intended_effect, atol=1e-5)
 
 
 @pytest.mark.parametrize('pre_turns,post_turns',

--- a/cirq/optimizers/decompositions_test.py
+++ b/cirq/optimizers/decompositions_test.py
@@ -235,8 +235,7 @@ def test_single_qubit_matrix_to_native_gates_known():
     cirq.testing.random_unitary(2) for _ in range(10)
 ])
 def test_single_qubit_matrix_to_native_gates_cases(intended_effect):
-    gates = cirq.single_qubit_matrix_to_phased_x_z(
-        intended_effect, atol=1e-6)
+    gates = cirq.single_qubit_matrix_to_phased_x_z(intended_effect, atol=1e-6)
     assert len(gates) <= 2
     assert_gates_implement_unitary(gates, intended_effect, atol=1e-5)
 

